src/coreclr/jit/targetriscv64.h

```diff
@@ -60,8 +60,12 @@
   #define CODE_ALIGN               4       // code alignment requirement
   #define STACK_ALIGN              16      // stack alignment requirement
 
+  #define FIRST_INT_CALLEE_SAVED  REG_S1
```

Comment:
Please note, that although `S1` is the first saved register, there is no `S2` after this register, but `A0` (in fact `a0-a7`). So iterating from `FIRST_INT_CALLEE_SAVED` to `LAST_INT_CALLEE_SAVED` may cause some problems. I'm not sure if this might be the case with this pull request, since I haven't had time to make a proper review, but I thought I'd write about it.

Reply:
Thanks
I will check it later.

Reply:
> Please note, that although `S1` is the first saved register, there is no `S2` after this register, but `A0` (in fact `a0-a7`). So iterating from `FIRST_INT_CALLEE_SAVED` to `LAST_INT_CALLEE_SAVED` may cause some problems.

This is Ok.
The related codes are here `CodeGen::genSaveCalleeSavedRegistersHelp` and `CodeGen::genRestoreCalleeSavedRegistersHelp`.
It doesn't require the S1 and S2 continous.

Reply:
I've looked at these functions and you are right. Thanks for checking!